pkg/stores/proxy/proxy_store_test.go

```diff
@@ -96,7 +96,13 @@ func receiveUntil(wc chan watch.Event, d time.Duration) error {
 			}
 
 			if event.Type == watch.Error {
-				return errors.New(fmt.Sprintf("watch chan should not have sent events of type [%s]", watch.Error))
+				status, ok := event.Object.(*metav1.Status)
```

Comment:
I don't think this test is working quite right for a few reasons:
- The goal of the `recieveUntil` function seems to be to receive events from the channel, and then return once it gets all of the expected events.
  - Right now, the return part of the function [returns](https://github.com/rancher/steve/blob/2365164708b72f5c10dd5a38799543a4c28aa793/pkg/stores/proxy/proxy_store_test.go#L114) once it has received all expected secrets, but doesn't check for all received errors.
  - This causes the function to return even if it hasn't received the error yet.
  -  You can fix this by changing the linked line to check if `errMsgs` is empty as well.
- Right now you are checking if you got the exact error message. But this won't work because [listAndWatch](https://github.com/rancher/steve/blob/2365164708b72f5c10dd5a38799543a4c28aa793/pkg/stores/proxy/proxy_store.go#L328) is wrapping the error message. 
  - I think if you change L103 to check for containment/the end of the error message, that should do the trick.

When combined, these issues seem to mean that the test isn't checking if it got an error.